[CORE-13745] rptest: add cloud topics to topic_creation_test.py

rockwotj opened the pull request:
- **rptest: add cloud topics to RapidTopicRecreateTest**
- **rptest: add cloud topics to TopicRecreateTest**

## Backports Required

- [x] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [ ] v25.2.x
- [ ] v25.1.x
- [ ] v24.3.x

## Release Notes

* none

rockwotj committed:
rptest: add cloud topics to RapidTopicRecreateTest

Since this test takes a while with the sleeps in there, we just have it
operate on two topics instead of parameterizing.

rockwotj committed:
rptest: add cloud topics to TopicRecreateTest

rockwotj commented:
/ci-repeat

rockwotj commented:
/ci-repeat

dotnwat commented on tests/rptest/tests/topic_creation_test.py:
```diff
@@ -185,6 +196,64 @@ def topic_is_healthy():
+            cloud_topics_enabled=True,
+        )
+
+        self.client().create_topic(spec)
```
See this comment on the same situation https://github.com/redpanda-data/redpanda/pull/28130#discussion_r2453651929

rockwotj replied:
Oh good catch thanks! Will update to rpk

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/74762-->
<!--670c9319-a558-46b9-b88e-13a777773c69-->
<!--title-->
## Retry command for [Build#74762](https://buildkite.com/redpanda/redpanda/builds/74762)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/shadow_linking_rnot_test.py::ShadowLinkingRandomOpsTest.test_node_operations@{"failures":true}
```
<!--/body-->

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--a0d54e75-eca3-4d37-a0a4-ccafb233110a-->
<!--TEST RESULTS-->
<!--9e979db2-3f6b-4c1b-80d5-8f62de9c2600-->
<!--TEST RESULTS-->
<!--07aa83d5-aeb9-43bc-b4c8-cd5c2c58daa7-->
<!--TEST RESULTS-->
<!--03777b1e-60d6-4a1e-acfd-69d32798af6c-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/74762">build#74762</a></summary>


| test_class                 | test_method               | test_arguments        | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                  | test_history                                                                                                                                                     |
|:---------------------------|:--------------------------|:----------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------|
| NodesDecommissioningTest   | test_recommissioning_node | {"cloud_topic": true} | integration | https://buildkite.com/redpanda/redpanda/builds/74762#019a0f06-9dc3-4b4b-9469-31e67a41daae | FLAKY         | 20/21    | upstream reliability is '98.6159169550173'. current run reliability is '95.23809523809523'. drift is 3.37782 and the allowed drift is set to 50. The test should PASS   | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=NodesDecommissioningTest&test_method=test_recommissioning_node |
| ShadowLinkingRandomOpsTest | test_node_operations      | {"failures": false}   | integration | https://buildkite.com/redpanda/redpanda/builds/74762#019a0f03-2d96-4ee1-a347-eee762fcd7b5 | FLAKY         | 6/21     | upstream reliability is '36.990595611285265'. current run reliability is '28.57142857142857'. drift is 8.41917 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingRandomOpsTest&test_method=test_node_operations    |
| ShadowLinkingRandomOpsTest | test_node_operations      | {"failures": false}   | integration | https://buildkite.com/redpanda/redpanda/builds/74762#019a0f06-9dbe-460c-9677-736f2cd4ffaa | FLAKY         | 7/21     | upstream reliability is '36.990595611285265'. current run reliability is '33.33333333333333'. drift is 3.65726 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingRandomOpsTest&test_method=test_node_operations    |
| ShadowLinkingRandomOpsTest | test_node_operations      | {"failures": true}    | integration | https://buildkite.com/redpanda/redpanda/builds/74762#019a0f03-2d97-412d-8538-7751212e57e1 | FLAKY         | 6/21     | upstream reliability is '53.84615384615385'. current run reliability is '28.57142857142857'. drift is 25.27473 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingRandomOpsTest&test_method=test_node_operations    |
| ShadowLinkingRandomOpsTest | test_node_operations      | {"failures": true}    | integration | https://buildkite.com/redpanda/redpanda/builds/74762#019a0f06-9dbf-480c-9740-3aebc0083fd7 | FLAKY         | 2/21     | upstream reliability is '62.37623762376238'. current run reliability is '9.523809523809524'. drift is 52.85243 and the allowed drift is set to 50. The test should FAIL | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingRandomOpsTest&test_method=test_node_operations    |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/74885">build#74885</a></summary>


| test_class             | test_method                                 | test_arguments   | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                    | test_history                                                                                                                                                                |
|:-----------------------|:--------------------------------------------|:-----------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:----------------------------------------------------------------------------------------------------------------------------------------------------------|:----------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| RapidTopicRecreateTest | test_topic_rapid_recreation                 | null             | integration | https://buildkite.com/redpanda/redpanda/builds/74885#019a1416-c6fa-4007-a90a-fb9481eeb385 | FLAKY         | 29/30    | upstream reliability is '100.0'. current run reliability is '95.23809523809523'. drift is 4.7619 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=RapidTopicRecreateTest&test_method=test_topic_rapid_recreation            |
| TopicRecreateTest      | test_cloud_topic_recreation_while_producing | null             | integration | https://buildkite.com/redpanda/redpanda/builds/74885#019a1416-c6fa-4007-a90a-fb9481eeb385 | FAIL          | 0/30     | The test has failed across all retries                                                                                                                    | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=TopicRecreateTest&test_method=test_cloud_topic_recreation_while_producing |
| TopicRecreateTest      | test_cloud_topic_recreation_while_producing | null             | integration | https://buildkite.com/redpanda/redpanda/builds/74885#019a1416-c700-4765-93b3-cb3f3daf6756 | FAIL          | 0/30     | The test has failed across all retries                                                                                                                    | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=TopicRecreateTest&test_method=test_cloud_topic_recreation_while_producing |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/75387">build#75387</a></summary>


| test_class                           | test_method                                 | test_arguments                                                                        | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                 | test_history                                                                                                                                                                  |
|:-------------------------------------|:--------------------------------------------|:--------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| EndToEndCloudTopicsTxTest            | test_write                                  | null                                                                                  | integration | https://buildkite.com/redpanda/redpanda/builds/75387#019a38c6-29a8-4802-a4dd-63fddcc7f73a | FLAKY         | 18/21    | upstream reliability is '92.7960927960928'. current run reliability is '85.71428571428571'. drift is 7.08181 and the allowed drift is set to 50. The test should PASS  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=EndToEndCloudTopicsTxTest&test_method=test_write                            |
| ShadowLinkConsumeGroupsMirroringTest | test_continuous_group_sync                  | {"source_cluster_spec": {"cluster_type": "redpanda"}, "with_failures": false}         | integration | https://buildkite.com/redpanda/redpanda/builds/75387#019a38c6-29ab-4fcc-b125-765813e207c6 | FLAKY         | 15/21    | upstream reliability is '100.0'. current run reliability is '71.42857142857143'. drift is 28.57143 and the allowed drift is set to 50. The test should PASS            | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkConsumeGroupsMirroringTest&test_method=test_continuous_group_sync |
| ShadowLinkingReplicationTests        | test_replication_timestamps_match           | {"source_cluster_spec": {"cluster_type": "redpanda"}, "timestamp_type": "CreateTime"} | integration | https://buildkite.com/redpanda/redpanda/builds/75387#019a38bf-7f8b-48c8-92f5-6d2f22399505 | FLAKY         | 20/21    | upstream reliability is '95.34883720930233'. current run reliability is '95.23809523809523'. drift is 0.11074 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingReplicationTests&test_method=test_replication_timestamps_match |
| ShadowLinkingReplicationTests        | test_replication_timestamps_match           | {"source_cluster_spec": {"cluster_type": "redpanda"}, "timestamp_type": "CreateTime"} | integration | https://buildkite.com/redpanda/redpanda/builds/75387#019a38c6-29a7-46a4-8e7f-4232227fc853 | FLAKY         | 20/21    | upstream reliability is '95.23809523809523'. current run reliability is '95.23809523809523'. drift is 0.0 and the allowed drift is set to 50. The test should PASS     | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingReplicationTests&test_method=test_replication_timestamps_match |
| MountUnmountIcebergTest              | test_simple_remount                         | {"cloud_storage_type": 1}                                                             | integration | https://buildkite.com/redpanda/redpanda/builds/75387#019a38bf-7f8b-48c8-92f5-6d2f22399505 | FLAKY         | 15/21    | upstream reliability is '77.12344280860702'. current run reliability is '71.42857142857143'. drift is 5.69487 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=MountUnmountIcebergTest&test_method=test_simple_remount                     |
| TopicRecreateTest                    | test_cloud_topic_recreation_while_producing | null                                                                                  | integration | https://buildkite.com/redpanda/redpanda/builds/75387#019a38c6-29a8-4802-a4dd-63fddcc7f73a | FLAKY         | 19/21    | upstream reliability is 'nan'. current run reliability is '90.47619047619048'. drift is nan and the allowed drift is set to 50. The test should FAIL                   | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=TopicRecreateTest&test_method=test_cloud_topic_recreation_while_producing   |
| WriteCachingFailureInjectionE2ETest  | test_crash_all                              | {"use_transactions": false}                                                           | integration | https://buildkite.com/redpanda/redpanda/builds/75387#019a38bf-7f8c-4f2b-828c-b840988d5b50 | FLAKY         | 16/21    | upstream reliability is '84.95007132667618'. current run reliability is '76.19047619047619'. drift is 8.7596 and the allowed drift is set to 50. The test should PASS  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=WriteCachingFailureInjectionE2ETest&test_method=test_crash_all              |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/75501">build#75501</a></summary>


| test_class                           | test_method                   | test_arguments                                                                                                                                                 | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                  | test_history                                                                                                                                                                  |
|:-------------------------------------|:------------------------------|:---------------------------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| EndToEndCloudTopicsTxTest            | test_write                    | null                                                                                                                                                           | integration | https://buildkite.com/redpanda/redpanda/builds/75501#019a4b97-f68c-4aaf-afd6-68f0ac62fb3f | FAIL          | 0/1      |                                                                                                                                                                         | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=EndToEndCloudTopicsTxTest&test_method=test_write                            |
| ShadowLinkConsumeGroupsMirroringTest | test_continuous_group_sync    | {"source_cluster_spec": {"cluster_type": "redpanda"}, "with_failures": false}                                                                                  | integration | https://buildkite.com/redpanda/redpanda/builds/75501#019a4b94-bdac-43a5-a735-bafedb6646cc | FLAKY         | 15/21    | upstream reliability is '93.06569343065694'. current run reliability is '71.42857142857143'. drift is 21.63712 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkConsumeGroupsMirroringTest&test_method=test_continuous_group_sync |
| MountUnmountIcebergTest              | test_simple_remount           | {"cloud_storage_type": 1}                                                                                                                                      | integration | https://buildkite.com/redpanda/redpanda/builds/75501#019a4b97-f68c-4fb3-a616-584a53de37e0 | FLAKY         | 15/21    | upstream reliability is '76.27737226277372'. current run reliability is '71.42857142857143'. drift is 4.8488 and the allowed drift is set to 50. The test should PASS   | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=MountUnmountIcebergTest&test_method=test_simple_remount                     |
| LogCompactionTxRemovalTest           | test_tx_control_batch_removal | null                                                                                                                                                           | integration | https://buildkite.com/redpanda/redpanda/builds/75501#019a4b94-bda4-4b42-bc27-a3a3c8d06b87 | FLAKY         | 12/21    | upstream reliability is '96.2233169129721'. current run reliability is '57.14285714285714'. drift is 39.08046 and the allowed drift is set to 50. The test should PASS  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=LogCompactionTxRemovalTest&test_method=test_tx_control_batch_removal        |
| TieredStorageTest                    | test_tiered_storage           | {"cloud_storage_type_and_url_style": [1, "virtual_host"], "test_case": {"name": "(TS_Read == True, TS_Timequery == True, SpilloverManifestUploaded == True)"}} | integration | https://buildkite.com/redpanda/redpanda/builds/75501#019a4b94-bda5-4cd4-bd3c-aafae5dc36d2 | FLAKY         | 20/21    | upstream reliability is '99.75961538461539'. current run reliability is '95.23809523809523'. drift is 4.52152 and the allowed drift is set to 50. The test should PASS  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=TieredStorageTest&test_method=test_tiered_storage                           |
</details>
<!--/body-->

rockwotj committed:
rptest: use rpk to create topics for topic_creation_test.py

This is because we can't enable cloud topics using the default client,
as it ignores unknown topic properties.

rockwotj commented:
/ci-repeat 1
skip-units
skip-redpanda-build
dt-repeat=10
tests/rptest/tests/topic_creation_test.py::RapidTopicRecreateTest
tests/rptest/tests/topic_creation_test.py::TopicRecreateTest.test_cloud_topic_recreation_while_producing

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/74885-->
<!--61efaff2-d623-4f04-a489-23b4420a0b0e-->
<!--ae78617b-817b-4ce9-8010-fee76e07de32-->
<!--title-->
## Retry command for [Build#74885](https://buildkite.com/redpanda/redpanda/builds/74885)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/topic_creation_test.py::TopicRecreateTest.test_cloud_topic_recreation_while_producing
```
<!--/body-->

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/75380-->
<!--c9dd8114-c680-460b-b674-29d1296ac711-->
<!--c2b293d2-af43-4817-a6ae-f9bd9cb3b01c-->
<!--a6e5c8b1-0b7f-47fb-bd68-9cd1ec47ae60-->
<!--1851e6d5-53f7-43aa-bad4-28033fad6839-->
<!--46ea614e-7b85-4ef3-96cc-ee17d0c8f29d-->
<!--059f9c69-d0de-41f7-8767-fa34bc79062e-->
<!--bdba8401-9c45-4266-86d5-f89fae6184e9-->
<!--87f18154-f9d6-48a0-9f40-61b9e492cbb1-->
<!--ee424f22-663c-45a1-ba7b-669dd7e4d2d0-->
<!--872cc493-5c9a-4130-81f6-7501838b73c7-->
<!--f4025a3d-83f9-470f-adf6-30a3a0420ed0-->
<!--27ecdb6c-c5f0-4cb8-adc6-0f618f1e3ad0-->
<!--title-->
## Retry command for [Build#75380](https://buildkite.com/redpanda/redpanda/builds/75380)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/topic_creation_test.py::TopicRecreateTest.test_topic_recreation_while_producing@{"cleanup_policy":"delete","workload":"ACKS_ALL"}
tests/rptest/tests/topic_creation_test.py::TopicRecreateTest.test_topic_recreation_while_producing@{"cleanup_policy":"compact","workload":"ACKS_ALL"}
tests/rptest/tests/topic_creation_test.py::TopicRecreateTest.test_topic_recreation_while_producing@{"cleanup_policy":"delete","workload":"IDEMPOTENT"}
tests/rptest/tests/topic_creation_test.py::TopicRecreateTest.test_topic_recreation_while_producing@{"cleanup_policy":"delete","workload":"ACKS_1"}
tests/rptest/tests/topic_creation_test.py::TopicRecreateTest.test_topic_recreation_while_producing@{"cleanup_policy":"compact","workload":"ACKS_1"}
tests/rptest/tests/topic_creation_test.py::TopicRecreateTest.test_cloud_topic_recreation_while_producing
tests/rptest/tests/topic_creation_test.py::TopicRecreateTest.test_topic_recreation_while_producing@{"cleanup_policy":"compact","workload":"IDEMPOTENT"}
```
<!--/body-->

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/75387-->
<!--13fe30e8-bc4a-47ad-9dd5-224a45201f4e-->
<!--title-->
## Retry command for [Build#75387](https://buildkite.com/redpanda/redpanda/builds/75387)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/topic_creation_test.py::TopicRecreateTest.test_cloud_topic_recreation_while_producing
```
<!--/body-->

rockwotj committed:
ct/l1/metastore: check abort source when purging

In the rapid topic creation test, it's seen that shutdown hangs with the
topic purger, and I think it is due to we see 100 topics we need to
purge, and with a timeout of 5 seconds each, it's probably pretty easy
to blow past our shutdown timeouts.

I wish our RPC system supported abort_sources.

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/75501-->
<!--59dcc190-3c99-4d05-ade7-fd99df47984e-->
<!--title-->
## Retry command for [Build#75501](https://buildkite.com/redpanda/redpanda/builds/75501)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/nodes_decommissioning_test.py::NodesDecommissioningTest.test_multiple_decommissions@{"cloud_topic":true}
```
<!--/body-->

rockwotj commented:
/ci-repeat 1
skip-units
skip-redpanda-build
dt-repeat=10
tests/rptest/tests/topic_creation_test.py::RapidTopicRecreateTest
tests/rptest/tests/topic_creation_test.py::TopicRecreateTest.test_cloud_topic_recreation_while_producing

rockwotj commented:
This is ready for a review now that I've fixed the shutdown hang
